jcl/src/java.base/share/classes/com/ibm/tools/attach/target/Attachment.java

```diff
@@ -343,6 +343,22 @@ String loadAgentLibrary(String agentLibrary, String options,
 				return Response.EXCEPTION_AGENT_INITIALIZATION_EXCEPTION
 						+ status;
 			}
+		} else {
+			Properties systemProperties = com.ibm.oti.vm.VM.getVMLangAccess().internalGetProperties();
+			String addr;
+			synchronized (systemProperties) {
+				addr = systemProperties.getProperty(LOCAL_CONNECTOR_ADDRESS);
+				if (Objects.isNull(addr)) {
+					addr = VMSupport.getAgentProperties().getProperty(LOCAL_CONNECTOR_ADDRESS);
+					if (!Objects.isNull(addr)) {
+						systemProperties.setProperty(LOCAL_CONNECTOR_ADDRESS, addr);
+					}
+				}
+			}
+			if (Objects.isNull(addr)) {
+				/* startLocalAgent() should have set the property. */
+				return Response.EXCEPTION_AGENT_INITIALIZATION_EXCEPTION + "loadAgentLibrary: " + LOCAL_CONNECTOR_ADDRESS + " not defined"; //$NON-NLS-1$ //$NON-NLS-2$
```

Comment:
Needs a space after EXCEPTION_AGENT_INITIALIZATION_EXCEPTION

Reply:
This code is the same as that in `startLocalAgent()`, correct?  If so, please factor it out into a separate static method.  While you are doing that, please add logging to record the agent address and any errors.